src/fillTank.test.js

```diff
@@ -1,11 +1,147 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+  const { fillTank } = require('./fillTank');
 
-  it('should ', () => {
+  it('should ', () => {});
 
+  // write tests here
+
+  it(`should be declared`, () => {
+    expect(fillTank).toBeInstanceOf(Function);
   });
 
-  // write tests here
+  it(`should return nothing`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+
+    expect(fillTank(customer)).toBeUndefined();
+  });
+
+  it(`should ordered full tank if amount is not given`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 10,
+      },
+    };
+
+    fillTank(customer, 5);
+
+    expect(customer).toEqual({
+      money: 2850,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 40,
+      },
+    });
+  });
+
+  it(`should not pour more than tank can accommodate`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 10,
+      },
+    };
+
+    fillTank(customer, 5, 50);
+
+    expect(customer).toEqual({
+      money: 2850,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 40,
+      },
+    });
+  });
+
+  it(`should not fill more than client can pay`, () => {
+    const customer = {
+      money: 50,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 10,
+      },
+    };
+
+    fillTank(customer, 5, 20);
```

Comment:
Use variables to represent reusable values like the price per liter, tank amount, and so on. This helps in maintaining the code and reduces hardcoding.

```suggestion
      const fuelPrice= 5;
      const fuelAmount= 40;
      
      fillTank(customer, fuelPrice, fuelAmount);
```